### DISCUSSION TITLE
关于心跳包发送的逻辑

### DISCUSSION THREAD

**Discussion by FishZe** [FishZe/go-bili-chat#22] [Ideas]
目前心跳包发送存在的痛点：
1. 多个直播间同时发送心跳，导致短时间的性能占用过高
2. 连接一个直播间就要单独开一个goroutine并用ticker每隔30秒发送，线程过多可能造成cpu调度问题影响性能

所以可能需要：
1. 将心跳包发送平均到30秒内，平均化减少突发
2. 使用一个或几个线程发送心跳包

所以，有几个选择：
1. 维持现有
2. 创建一个goroutine，每间隔30/N (N为当前连接数)发送一个直播间的心跳包，但是可能对于极大的连接量可能存在负优化
3. 创建一个SetHeartBeatPolicy让开发者自行选择
4. 其他，请评论

**Comment by lhz226** [comment]
3，保留现有方法并尝试方法2，提供自由选择

**Comment by XiaoMiku01** [comment]
3 现在用的还行，新方法慢慢测试吧
